src/main/kotlin/de/danielmaile/mpp/block/BlockBreakingService.kt

```diff
@@ -66,30 +63,30 @@ private val MINING_FATIGUE = MobEffectInstance(MobEffect.byId(4)!!, Integer.MAX_
 
 object BlockBreakingService {
 
-    fun init() {
-        // listen to player action packets
-        ProtocolLibrary.getProtocolManager().addPacketListener(object :
-                PacketAdapter(inst(), ListenerPriority.NORMAL, PacketType.Play.Client.BLOCK_DIG) {
-                override fun onPacketReceiving(event: PacketEvent) {
-                    val sequence = event.packet.integers.read(0)
-                    val blockPosition = event.packet.blockPositionModifier.values[0]
-                    val digType = event.packet.playerDigTypes.values[0]
-
-                    if (digType == EnumWrappers.PlayerDigType.START_DESTROY_BLOCK) {
-                        val block = event.player.world.getBlockAt(blockPosition.x, blockPosition.y, blockPosition.z)
-                        if (block.isCustom()) {
-                            damageBlock(block, event.player, sequence)
-                        }
-                    }
+    @PacketListener(ignoreCancelled = true)
+    fun onBlockDig(event: de.danielmaile.mpp.packet.PacketEvent<ServerboundPlayerActionPacket>) {
+        val sequence = event.packet.sequence
+        val blockPos = event.packet.pos
 
-                    if (digType == EnumWrappers.PlayerDigType.STOP_DESTROY_BLOCK || digType == EnumWrappers.PlayerDigType.ABORT_DESTROY_BLOCK) {
-                        val block = event.player.world.getBlockAt(blockPosition.x, blockPosition.y, blockPosition.z)
-                        stopDamaging(block)
-                    }
+        when (event.packet.action) {
+            Action.START_DESTROY_BLOCK -> {
+                val block = event.player.world.getBlockAt(blockPos.x, blockPos.y, blockPos.z)
+                if (block.isCustom()) {
+                    damageBlock(block, event.player, sequence)
                 }
-            })
+                event.player.sendMessage("You're breaking a block!")
+            }
+            Action.STOP_DESTROY_BLOCK, Action.ABORT_DESTROY_BLOCK -> {
+                val block = event.player.world.getBlockAt(blockPos.x, blockPos.y, blockPos.z)
+                stopDamaging(block)
+            }
+            else -> {
+                return
+            }
+        }
+    }
 
-        // handle ticks
+    fun initializeBreakingScheduler() {
```

Comment:
Should we call this function init? I think thats common practice in objects

Reply:
init doesn't give a great perception of what the method does, although we can call it init and it would only take a ⌘B to know what it does, I believe a good name is way better than having to read through the code to understand a method. What do you think about that?

Reply:
And way better than having a long description.

Reply:
LIke I said before I would use an init function because that way you know why this is called in the onEnable Method. You could create an init function that calls your function. I can live with both.